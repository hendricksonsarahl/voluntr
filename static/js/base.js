// display error messages as Bootstrap alert elements in browser
// Bootstrap will handle closing them
function displayError(message) {

  //allow for direct input of error objects:
  if (message instanceof Error) {
    message = message.message;
  }

  //select a container for the alert
  var alertContainer = document.querySelector(".alert-container");

  //build a new DOM element
  var alert = document.createElement("div");
  alert.className = `alert alert-danger alert-dismissible fade in`;
  alert.innerHTML = `<button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button><strong>Error: </strong>${message}`;

  //append the new DOM element to the container
  alertContainer.appendChild(alert);
}

// Sign out of Google connection, then redirect to "/"
function signOut(event) {
  //don't immediately redirect the user
  event.preventDefault();

  //send sign-out request to Google
  var auth2 = gapi.auth2.getAuthInstance();
  auth2.signOut().then(function() {
    console.log("User signed out.");

    //redirect user after confirmation of sign-out received
    window.location.href = "/";
  });
}

//don't run this outside of a browser environment (e.g., when testing in Node)
if (typeof window !== "undefined") {
  
  // Code in this anonymous function is immediately invoked once this script loads:
  (function() {

    //when the sign-out link is clicked, the signOut function is executed
    if (window.location.pathname.substring(0,4) === "/org") {
      var signOutLink = document.getElementById("signOut");
      signOutLink.addEventListener("click", signOut);
    }
    
  })();
}

// Export all named, top-level functions for use in unit tests.
// The leading if statement should prevent this block from running in the browser.
if (typeof module !== "undefined" && module.exports) {
  module.exports = {
    displayError: displayError,
    signOut: signOut
  };
}