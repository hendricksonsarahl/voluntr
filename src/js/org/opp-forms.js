// Update form to require/not allow date and times, based on Flexible Schedule checkbox
export function toggleFlexible(e) {
  const scheduleInputs = document.querySelectorAll(".schedule-input");
  const scheduleLabels = document.querySelectorAll(".schedule-label");
  const addressInput = document.getElementById("address");

  if (e.target.checked) {
    // Dim labels and inputs, stop requiring them, mark input disabled
    scheduleLabels.forEach(label => {
      label.classList.add("text-muted");
    });
    scheduleInputs.forEach(input => {
      input.removeAttribute("required");
      input.setAttribute("disabled", true);
    });

    // Focus the address input
    addressInput.focus();
  } else {
    // Reset labels and inputs to initial state
    scheduleLabels.forEach((label)=> {
      label.classList.remove("text-muted");
    });
    scheduleInputs.forEach((input)=> {
      input.setAttribute("required", true);
      input.removeAttribute("disabled");
    });
  }
}