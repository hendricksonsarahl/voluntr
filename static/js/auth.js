// TODO: Finish this function
function onSignIn(googleUser) {
  // Build an object with the data we wish to send to server
  var profile = googleUser.getBasicProfile();
  var authToken = googleUser.getAuthResponse().id_token;
  var userDataToSend = {
    contactName: profile.getName(),
    email: profile.getEmail(),
    authToken: authToken
  };

  // build a fetch request to POST user's Google data to server for verification:
  var authUrl = "/org/login";
  var requestOptions = {
    method: "POST",
    body: JSON.stringify(userDataToSend)
  };

  requestOptions.headers = new Headers();
  requestOptions.headers.append("Content-Type", "application/json");

  // Send the request, and respond to both successful and failure outcomes:
  fetch(authUrl, requestOptions)
    .then(function(response) {
      if (response.status >= 300) {
        displayError(response);
      } else {
        response
          .json()
          .then(function(responseData) {
            // TODO: Should request more data from new users, and redirect to logged-in view for returning users.
            console.log("Success! Server responded with: ", responseData);
            showSignUpForm();
          })
          .catch(displayError);
      }
    })
    .catch(displayError);
}

function signOut(event) {
  //don't immediately redirect the user
  event.preventDefault();

  //send sign-out request to Google
  var auth2 = gapi.auth2.getAuthInstance();
  auth2.signOut().then(function() {
    console.log("User signed out.");

    //redirect user after confirmation of sign-out received
    window.location.href = "/";
  });
}

function showSignUpForm() {
  var signUpForm = document.querySelector('.signup-row');
  signUpForm.classList.remove("hidden");
}

//don't run this outside of a browser environment (e.g., when testing in Node)
if (typeof window !== "undefined") {
  // Code in this anonymous function is immediately invoked once this script loads:
  (function() {
    //when the sign-out link is clicked, the signOut function is executed
    var signOutLink = document.getElementById("signOut");
    signOutLink.addEventListener("click", signOut);
  })();
}

// Export all named, top-level functions for use in unit tests.
// The leading if statement should prevent this block from running in the browser.
if (typeof module !== "undefined" && module.exports) {
  module.exports = {
    onSignIn: onSignIn,
    signOut: signOut,
    showSignUpForm: showSignUpForm
  };
}
