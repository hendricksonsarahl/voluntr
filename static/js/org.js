// ~~~~~~~~~~~~~~~~~~~Misc helper functions~~~~~~~~~~~~~~~~~~

// display error messages as Bootstrap alert elements in browser
// Bootstrap will handle closing them
function displayError(message) {

  //allow for direct input of error objects:
  if (message instanceof Error) {
    message = message.message;
  }

  //select a container for the alert
  var alertContainer = document.querySelector(".alert-container");

  //build a new DOM element
  var alert = document.createElement("div");
  alert.className = `alert alert-danger alert-dismissible fade in`;
  alert.innerHTML = `<button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button><strong>Error: </strong>${message}`;

  //append the new DOM element to the container
  alertContainer.appendChild(alert);
}

// delete a cookie by setting its expiration date in the past
function deleteCookie(name) {
  document.cookie = name +'=; Path=/; Expires=Thu, 01 Jan 1970 00:00:01 GMT;';
}

// ~~~~~~~~~~~~~~~~~Add/Edit Opp form~~~~~~~~~~~~~~~~~

// Update form to require/not allow date and times, based on Flexible Schedule checkbox
function toggleFlexible(e) {
  var scheduleInputs = document.querySelectorAll('.schedule-input');
  var scheduleLabels = document.querySelectorAll('.schedule-label');
  var addressInput = document.getElementById('address');

  if(e.target.checked) {

    // Dim labels and inputs, stop requiring them, mark input disabled
    scheduleLabels.forEach(function(label){
      label.classList.add('text-muted');
    });
    scheduleInputs.forEach(function(input){
      input.removeAttribute("required");
      input.setAttribute("disabled", true);
    });

    // Focus the address input
    addressInput.focus();
  } else {

    // Reset labels and inputs to initial state
    scheduleLabels.forEach(function(label){
      label.classList.remove('text-muted');
    });
    scheduleInputs.forEach(function(input){
      input.setAttribute("required", true);
      input.removeAttribute("disabled");
    });
  }
}


// ~~~~~~~~~~~~~~~~User account authorization:~~~~~~~~~~~~~~~~~~~
// Callback function for Google API's sign-in event
function onSignIn(googleUser) {

  // Only proceed if called from login page:
  if (window.location.pathname !== "/org/login") {
    return false;
  }
  
  var profile = googleUser.getBasicProfile();
  var authToken = googleUser.getAuthResponse().id_token;

  // Build an object with the data we wish to send to server 
  var userDataToSend = {
    authToken: authToken
  };

  // build a fetch request to POST user's Google data to server for verification:
  var authUrl = "/org/login";
  var requestOptions = {
    method: "POST",
    body: JSON.stringify(userDataToSend)
  };

  requestOptions.headers = new Headers();
  requestOptions.headers.append("Content-Type", "application/json");

  // Send the request, and respond to both successful and failure outcomes:
  fetch(authUrl, requestOptions)
    .then(function(response) {
      if (response.status >= 300) {
        displayError(response);
      } else {
        response
          .json()
          .then(function(responseData) {
            
            // Based on token validity, and Voluntr account existence, either:
            // Redirect to logged-in view, show sign-up form, or show an error message
            if(responseData.valid_token) {
              if (responseData.account_exists) {

                //add token to cookie, then redirect to logged-in view
                document.cookie = `token=${responseData.token}; path=/`;
                redirectToOrgHome();
              } else {
                showSignUpForm(profile.getName(), profile.getEmail(), authToken);
              }
            } else {
              throw Error('Received invalid Google authorization token.')
            }
          })
          .catch(displayError);
      }
    })
    .catch(displayError);
}

// Executed after successful Google sign-in, with no existing Voluntr account:
function showSignUpForm(contactName, email, token) {
  var signUpForm = document.querySelector('.signup-row');
  var googleDiv = document.querySelector('.g-signin2');
  var contactNameInput = document.getElementById('contactName');
  var emailInput = document.getElementById('email');
  var tokenInput = document.getElementById('token');

  //pre-populate Contact Name and Contact Email fields with data from Google account:
  contactNameInput.value = contactName;
  emailInput.value = email;
  tokenInput.value = token;

  //display the form, hide the Google sign-in button
  signUpForm.classList.remove("hidden");
  googleDiv.classList.add("hidden");
}

// Executed after successful Google sign-in to an existing Voluntr account:
function redirectToOrgHome() {
  var redirectRow = document.querySelector('.redirect-row');
  var googleDiv = document.querySelector('.g-signin2');

  redirectRow.classList.remove("hidden");
  googleDiv.classList.add("hidden");
  
  setTimeout(function() {
    window.location.href="/org/opportunities";    
  }, 1200);
}

// Sign out of Google connection, delete token cookie, then redirect to "/"
function signOut(event) {
  //don't immediately redirect the user
  event.preventDefault();

  //send sign-out request to Google
  var auth2 = gapi.auth2.getAuthInstance();
  auth2.disconnect();
  auth2.signOut().then(function() {

    //delete the OAuth token from browser cookies:
    deleteCookie('token');

    //redirect user after confirmation of sign-out received
    window.location.href = "/";
  });
}


//don't run this outside of a browser environment (e.g., when testing in Node)
if (typeof window !== "undefined") {
  
  // Code in this anonymous function is immediately invoked once this script loads:
  (function() {

    //when the sign-out link is clicked, the signOut function is executed
    var signOutLink = document.querySelectorAll(".signOutLink");
    var flexibleCheckbox = document.getElementById("flexible");

    if (signOutLink) {
      signOutLink.forEach(function(link){
        link.addEventListener("click", signOut);
      });
    }

    if (flexibleCheckbox) {
      flexibleCheckbox.addEventListener("click", toggleFlexible);
    }
    
  })();
}

// Export all named, top-level functions for use in unit tests.
// The leading if statement should prevent this block from running in the browser.
if (typeof module !== "undefined" && module.exports) {
  module.exports = {
    displayError: displayError,
    signOut: signOut,
    onSignIn: onSignIn,
    showSignUpForm: showSignUpForm,
    redirectToOrgHome: redirectToOrgHome
  };
}