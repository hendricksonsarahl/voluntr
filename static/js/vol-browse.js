// returns true if the page's opportunity is in the provided array
function isSaved(arr) {
  function checkId(obj) {

    //window.currentOpp is a global variable loaded in the HTML created by the Jinja template
    return obj.id === window.currentOpp.id;
  }

  return arr.findIndex(checkId) > -1
}

// return the savedOpps value in localStorage. if it doesn't exist yet, create an empty array first
function loadStore() {
  //get the current stored value for savedOpps
  var store = JSON.parse(localStorage.getItem("savedOpps"));

  //if no local store has been created for savedOpps, create an empty array to hold it
  if(!store) {
    var newStore = JSON.stringify([])
    localStorage.setItem("savedOpps", newStore);
    store = JSON.parse(localStorage.getItem("savedOpps"));
  }

  return store;
}

// add opportunities to a JSON-encoded localStorage array
function saveOpportunity(store){

  //check to see if the opportunity has already been saved:
  if(!isSaved(store)) {

    //add the page's currentOpp to the savedOpps store
    store.push(window.currentOpp);
    localStorage.setItem("savedOpps", JSON.stringify(store));
  }
}

// change the appearance and text of the save button if the opp has been saved
function updateSaveButton() {
  this.classList.remove('btn-default')
  this.classList.add('btn-success');
  this.innerHTML = 'Saved!&nbsp;<span class="glyphicon glyphicon-ok"></span>';  
}

//don't run this outside of a browser environment (e.g., when testing in Node)
if (typeof window !== "undefined") {
  
  // Code in this anonymous function is immediately invoked once this script loads:
  (function() {
    var store = loadStore();
    var saveButton = document.getElementById('save-button');
    
    // update the save button if the opp is already saved
    if (isSaved(store)) {
      updateSaveButton.bind(saveButton)();
    }

    // on Save click, add the opp to localStorage and change button appearance
    saveButton.addEventListener('click', function(e){
      e.preventDefault();
      saveOpportunity(store);
      updateSaveButton.bind(this)();
    });

  })();
}

// Export all named, top-level functions for use in unit tests.
// The leading if statement should prevent this block from running in the browser.
if (typeof module !== "undefined" && module.exports) {
  module.exports = {
    saveOpportunity: saveOpportunity
  };
}