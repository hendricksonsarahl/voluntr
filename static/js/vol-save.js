// return the savedOpps value in localStorage. if it doesn't exist yet, create an empty array first
function loadStore() {
  //get the current stored value for savedOpps
  var store = JSON.parse(localStorage.getItem("savedOpps"));

  //if no local store has been created for savedOpps, create an empty array to hold it
  if (!store) {
    store = [];
  }

  return store;
}

// map the opportunities found in localStorage to HTML panels, then append them to a container
function showOpps(store, parentElt) {
  var headerElt = document.createElement("h2");
  headerElt.textContent = "My Saved Posts";
  parentElt.appendChild(headerElt);

  store.forEach(function(opp) {
    var oppElt = document.createElement("div");
    oppElt.innerHTML = `
    <div class="panel panel-default" data-id="${opp.id}">
      <div class="panel-body">
        <h4>${opp.title}</h4>
        <div class="pull-right">
          <a class="btn btn-primary" href="#" role="button">View&nbsp;
            <div class="glyphicon glyphicon-play"></div>
          </a>
          <a class="btn btn-danger remove-button" href="#" role="button">Remove&nbsp;
            <div class="glyphicon glyphicon-remove"></div>
          </a>
        </div>
      </div>
    </div>
    `;
    parentElt.appendChild(oppElt);
  });
}

// Display message if no opportunities have been saved yet
function showNoOppsMessage(parentElt) {
  parentElt.innerHTML =
    '<h2>Nothing Saved Yet</h2><p><a href="/opportunities">Continue browsing opportunities</a></p>';
}

// Remove an opportunity from localStorage
function removeOppFromStore(oppId) {
  var oppList = loadStore();

  // Find the index of the target opportunity in the store:
  var indexToRemove = oppList.findIndex(function(opp) {
    return opp.id == oppId;
  });

  // Remove the opportunity from the store and save to localStorage
  oppList.splice(indexToRemove, 1);
  localStorage.setItem("savedOpps", JSON.stringify(oppList));
}

// Remove an opportunity's panel from the page
function removeOppFromPage(oppPanel) {

  //Remove the opportunity's panel from this page:
  oppPanel.parentNode.removeChild(oppPanel);
}

// Update the display
function render(store, parentElt) {

  // Remove everything currently in parentElt
  parentElt.innerHTML = "";

  // Show saved opps as panels if they exist; otherwise show a message
  if (store.length > 0) {
    showOpps(store, parentElt);
  } else {
    showNoOppsMessage(parentElt);
  }
}

//don't run this outside of a browser environment (e.g., when testing in Node)
if (typeof window !== "undefined") {
  // Code in this anonymous function is immediately invoked once this script loads:
  (function() {
    var store = loadStore();
    var oppListParent = document.getElementById("opp-container");

    //initial render
    render(store, oppListParent);

    //listen for clicks on Remove buttons
    oppListParent.addEventListener("click", function(e) {
      if (
        e.target.classList.contains("remove-button") ||
        e.target.parentNode.classList.contains("remove-button")
      ) {
        var clickedOppId = e.target.parentNode.parentNode.parentNode.dataset.id
        removeOppFromStore(clickedOppId);

        //re-load the store from localStorage, and re-render the view
        store = loadStore();
        render(store, oppListParent);
      }
    });
  })();
}

// Export all named, top-level functions for use in unit tests.
// The leading if statement should prevent this block from running in the browser.
if (typeof module !== "undefined" && module.exports) {
  module.exports = {};
}
