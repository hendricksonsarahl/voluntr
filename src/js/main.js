import { onSignIn, signOut } from "./org/auth";
import { toggleFlexible } from "./org/opp-forms";
import { idInArray } from "./vol/helpers";
import {
  loadStore,
  saveOpportunity,
  removeOppFromStore
} from "./vol/local-storage";
import {
  selectAllCategories,
  toggleSelectAllCheckbox
} from "./vol/filters-form";
import { updateSaveButton, renderSavedOpps } from "./vol/dom-saved-opps";

// Webpack checks uses file as entry point for bundling CSS assets
import "../../node_modules/bootswatch/cosmo/bootstrap.min.css";
import "../../node_modules/intro.js/introjs.css";
import "../css/main.css";

// don't run this outside of a browser environment (e.g., when testing in Node)
if (typeof window !== "undefined") {
  // Code in this anonymous function is immediately invoked once this script loads:
  (() => {
    // get data from localStorage:
    let store = loadStore();

    // create variables from DOM elements
    // if a given element isn't found on the current page, its variable has value of null
    const signOutLink = document.querySelectorAll(".signOutLink");
    const flexibleCheckbox = document.getElementById("flexible");
    const selectAllButton = document.getElementById("selectAll");
    const categoryInputs = document.querySelectorAll("input[name=category]");
    const saveButton = document.getElementById("save-button");
    const oppListParent = document.getElementById("opp-container");
    const tutorialButton = document.getElementById("tutorial-button");

    // Give the onSignIn callback function a global scope, so it can be accessed by Google's gapi script
    window.onSignIn = onSignIn;

    // start the Intro JS tour when Tutorial button is clicked
    if (tutorialButton) {
      tutorialButton.addEventListener("click", () => {
        window.introJs.introJs().setOption("showProgress", true).start();
      });
    }

    // when the sign-out link is clicked, the signOut function is executed
    if (signOutLink) {
      signOutLink.forEach(link => {
        link.addEventListener("click", signOut);
      });
    }

    // set up Add/Edit opp forms:
    if (flexibleCheckbox) {
      flexibleCheckbox.addEventListener("click", toggleFlexible);
    }

    // set up Filters page:
    if (selectAllButton) {
      selectAllButton.addEventListener("change", selectAllCategories);
      categoryInputs.forEach(cat => {
        cat.addEventListener("change", toggleSelectAllCheckbox);
      });
    }

    // set up Browse Opportunities page:
    if (saveButton) {
      // update the save button if the opp is already saved on page load
      // (currentOpp is a global variable loaded in the HTML created by the Jinja template)

      if (idInArray(store, currentOpp)) {
        updateSaveButton.bind(saveButton)(true);
      }

      // on Save Button click, add the opp to localStorage and change button appearance
      saveButton.addEventListener("click", e => {
        // depending on if the opportunity has already been saved, either save or remove it:
        if (!idInArray(store, currentOpp)) {
          saveOpportunity(store, currentOpp);
          updateSaveButton.bind(e.target)(true);
          store = loadStore();
        } else {
          removeOppFromStore(currentOpp.id);
          updateSaveButton.bind(e.target)(false);
          store = loadStore();
        }
      });
    }

    // set up My Saved Opportunities page:
    if (oppListParent) {
      // initial render
      renderSavedOpps(store, oppListParent);

      // listen for clicks on Remove buttons
      oppListParent.addEventListener("click", e => {
        if (
          e.target.classList.contains("remove-button") ||
          e.target.parentNode.classList.contains("remove-button")
        ) {
          const clickedOppId =
            e.target.parentNode.parentNode.parentNode.parentNode.dataset.id;
          removeOppFromStore(clickedOppId);

          // re-load the store from localStorage, and re-render the view
          store = loadStore();
          renderSavedOpps(store, oppListParent);
        }
      });
    }
  })();
}
