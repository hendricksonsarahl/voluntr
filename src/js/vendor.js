// Gather vendor packages for bundling by Webpack

window.$ = require("jquery");
window.jQuery = require("jquery");

// Import all of Bootstrap:
// require("bootstrap");

// Import Bootstrap JS modules as needed (saves space):
require('bootstrap/js/transition.js')
require('bootstrap/js/alert.js')
// require('bootstrap/js/button.js')
// require('../../js/carousel.js')
require('bootstrap/js/collapse.js')
require('bootstrap/js/dropdown.js')
// require('../../js/modal.js')
// require('../../js/tooltip.js')
// require('../../js/popover.js')
// require('../../js/scrollspy.js')
// require('../../js/tab.js')
// require('../../js/affix.js')

window.introJs = require("intro.js");