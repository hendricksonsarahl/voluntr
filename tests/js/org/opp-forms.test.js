import { toggleFlexible } from "../../../src/js/org/opp-forms.js";

let addressInput;
const setUpDOM = () => {
  document.body.innerHTML = `
    <div class="row">
      <div class="col-xs-12 col-sm-4">
        <label for="date">
          <h3>Flexible Schedule?</h3>
        </label>
      </div>
      <div class="col-xs-12 col-sm-8">
        <div class="checkbox checkbox-opp-form">
          <label>
            <input id="flexible" name="flexible" type="checkbox" value="on">
            Yes, please display "Flexible schedule" for this post instead of a specific date and time.
          </label>
        </div>
      </div>
    </div>
    <br />

    <div class="row">
      <div class="col-xs-12 col-sm-4">
        <label for="date" class="schedule-label">
          <h3>Date</h3>
        </label>
      </div>
      <div class="col-xs-12 col-sm-8">
        <input class="form-control schedule-input" id="date" name="date" type="date" required>
      </div>
    </div>
    <br />
    <br class="hidden-xs" />

    <div class="row">
      <div class="col-xs-12 col-sm-4">
        <label for="startTime" class="schedule-label">
          <h3>Start Time</h3>
        </label>
      </div>
      <div class="col-xs-12 col-sm-8">
        <input class="form-control schedule-input" id="startTime" type="time" name="startTime" value="12:00" required>
      </div>
    </div>
    <br />
    <br class="hidden-xs" />

    <div class="row">
      <div class="col-xs-12 col-sm-4">
        <label for="endTime" class="schedule-label">
          <h3>End Time</h3>
        </label>
      </div>
      <div class="col-xs-12 col-sm-8">
        <input class="form-control schedule-input" id="endTime" type="time" name="endTime" value="12:30" required>
      </div>
    </div>
    <br />
    <br class="hidden-xs" />

    <div class="row">
      <div class="col-xs-12 col-sm-4">
        <label for="address">
          <h3>Street Address</h3>
        </label>
      </div>
      <div class="col-xs-12 col-sm-8">
        <input class="form-control" id="address" type="text" name="address" maxlength="120" required>
      </div>
    </div>
    <br />
    <br class="hidden-xs" />
  `;

  addressInput = document.getElementById("address");
};

beforeEach(setUpDOM);

describe("toggleFlexible", () => {
  it("Rendered body HTML matches snapshot for e.target.checked=true", () => {
    const clickEvent = { target: { checked: true } };
    toggleFlexible(clickEvent);

    const renderedHTML = document.body.innerHTML;
    expect(renderedHTML).toMatchSnapshot();
  });

  it("Address input has focus when e.target.checked=true", () => {
    const clickEvent = { target: { checked: true } };
    toggleFlexible(clickEvent);

    expect(document.activeElement).toEqual(addressInput);
  });

  it("Rendered body HTML matches snapshot for e.target.checked=false", () => {
    const clickEvent = { target: { checked: false } };
    toggleFlexible(clickEvent);

    const renderedHTML = document.body.innerHTML;
    expect(renderedHTML).toMatchSnapshot();
  });

  // it("Rendered button HTML matches snapshot for 'false' argument", () => {
  //   updateSaveButton.bind(saveButton, false)();
  //   const renderedHTML = saveButton.outerHTML;
  //   expect(renderedHTML).toMatchSnapshot();
  // });
});
