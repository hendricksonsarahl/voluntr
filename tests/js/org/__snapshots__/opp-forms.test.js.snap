// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggleFlexible Rendered body HTML matches snapshot for e.target.checked=false 1`] = `
"
    <div class=\\"row\\">
      <div class=\\"col-xs-12 col-sm-4\\">
        <label for=\\"date\\">
          <h3>Flexible Schedule?</h3>
        </label>
      </div>
      <div class=\\"col-xs-12 col-sm-8\\">
        <div class=\\"checkbox checkbox-opp-form\\">
          <label>
            <input id=\\"flexible\\" name=\\"flexible\\" type=\\"checkbox\\" value=\\"on\\">
            Yes, please display \\"Flexible schedule\\" for this post instead of a specific date and time.
          </label>
        </div>
      </div>
    </div>
    <br>

    <div class=\\"row\\">
      <div class=\\"col-xs-12 col-sm-4\\">
        <label for=\\"date\\" class=\\"schedule-label\\">
          <h3>Date</h3>
        </label>
      </div>
      <div class=\\"col-xs-12 col-sm-8\\">
        <input class=\\"form-control schedule-input\\" id=\\"date\\" name=\\"date\\" type=\\"date\\" required=\\"true\\">
      </div>
    </div>
    <br>
    <br class=\\"hidden-xs\\">

    <div class=\\"row\\">
      <div class=\\"col-xs-12 col-sm-4\\">
        <label for=\\"startTime\\" class=\\"schedule-label\\">
          <h3>Start Time</h3>
        </label>
      </div>
      <div class=\\"col-xs-12 col-sm-8\\">
        <input class=\\"form-control schedule-input\\" id=\\"startTime\\" type=\\"time\\" name=\\"startTime\\" value=\\"12:00\\" required=\\"true\\">
      </div>
    </div>
    <br>
    <br class=\\"hidden-xs\\">

    <div class=\\"row\\">
      <div class=\\"col-xs-12 col-sm-4\\">
        <label for=\\"endTime\\" class=\\"schedule-label\\">
          <h3>End Time</h3>
        </label>
      </div>
      <div class=\\"col-xs-12 col-sm-8\\">
        <input class=\\"form-control schedule-input\\" id=\\"endTime\\" type=\\"time\\" name=\\"endTime\\" value=\\"12:30\\" required=\\"true\\">
      </div>
    </div>
    <br>
    <br class=\\"hidden-xs\\">

    <div class=\\"row\\">
      <div class=\\"col-xs-12 col-sm-4\\">
        <label for=\\"address\\">
          <h3>Street Address</h3>
        </label>
      </div>
      <div class=\\"col-xs-12 col-sm-8\\">
        <input class=\\"form-control\\" id=\\"address\\" type=\\"text\\" name=\\"address\\" maxlength=\\"120\\" required=\\"\\">
      </div>
    </div>
    <br>
    <br class=\\"hidden-xs\\">
  "
`;

exports[`toggleFlexible Rendered body HTML matches snapshot for e.target.checked=true 1`] = `
"
    <div class=\\"row\\">
      <div class=\\"col-xs-12 col-sm-4\\">
        <label for=\\"date\\">
          <h3>Flexible Schedule?</h3>
        </label>
      </div>
      <div class=\\"col-xs-12 col-sm-8\\">
        <div class=\\"checkbox checkbox-opp-form\\">
          <label>
            <input id=\\"flexible\\" name=\\"flexible\\" type=\\"checkbox\\" value=\\"on\\">
            Yes, please display \\"Flexible schedule\\" for this post instead of a specific date and time.
          </label>
        </div>
      </div>
    </div>
    <br>

    <div class=\\"row\\">
      <div class=\\"col-xs-12 col-sm-4\\">
        <label for=\\"date\\" class=\\"schedule-label text-muted\\">
          <h3>Date</h3>
        </label>
      </div>
      <div class=\\"col-xs-12 col-sm-8\\">
        <input class=\\"form-control schedule-input\\" id=\\"date\\" name=\\"date\\" type=\\"date\\" disabled=\\"true\\">
      </div>
    </div>
    <br>
    <br class=\\"hidden-xs\\">

    <div class=\\"row\\">
      <div class=\\"col-xs-12 col-sm-4\\">
        <label for=\\"startTime\\" class=\\"schedule-label text-muted\\">
          <h3>Start Time</h3>
        </label>
      </div>
      <div class=\\"col-xs-12 col-sm-8\\">
        <input class=\\"form-control schedule-input\\" id=\\"startTime\\" type=\\"time\\" name=\\"startTime\\" value=\\"12:00\\" disabled=\\"true\\">
      </div>
    </div>
    <br>
    <br class=\\"hidden-xs\\">

    <div class=\\"row\\">
      <div class=\\"col-xs-12 col-sm-4\\">
        <label for=\\"endTime\\" class=\\"schedule-label text-muted\\">
          <h3>End Time</h3>
        </label>
      </div>
      <div class=\\"col-xs-12 col-sm-8\\">
        <input class=\\"form-control schedule-input\\" id=\\"endTime\\" type=\\"time\\" name=\\"endTime\\" value=\\"12:30\\" disabled=\\"true\\">
      </div>
    </div>
    <br>
    <br class=\\"hidden-xs\\">

    <div class=\\"row\\">
      <div class=\\"col-xs-12 col-sm-4\\">
        <label for=\\"address\\">
          <h3>Street Address</h3>
        </label>
      </div>
      <div class=\\"col-xs-12 col-sm-8\\">
        <input class=\\"form-control\\" id=\\"address\\" type=\\"text\\" name=\\"address\\" maxlength=\\"120\\" required=\\"\\">
      </div>
    </div>
    <br>
    <br class=\\"hidden-xs\\">
  "
`;
